iam/__tests__/githubForkedRepoProvider.test.ts

```diff
@@ -0,0 +1,223 @@
+      `https://github.com/login/oauth/access_token?client_id=${clientId}&client_secret=${clientSecret}&code=${code}`,
+      {},
+      {
+        headers: { Accept: "application/json" },
+      }
+    );
+
+    // Check the request to get the user
+    expect(mockedAxios.get).toBeCalledWith("https://api.github.com/user", {
+      headers: { Authorization: "token 762165719dhiqudgasyuqwt6235" },
+    });
+
+    // Check the request to get the repo
+    expect(mockedAxios.get).toBeCalledWith(`https://api.github.com/users/${validGithubUserResponse.data.login}/repos`, {
+      headers: { Authorization: "token 762165719dhiqudgasyuqwt6235" },
+    });
+
+    expect(forkedGithubRepoProviderPayload).toEqual({
+      valid: true,
+      record: {
+        id: `${validGithubUserResponse.data.id}gte1Fork`,
+      },
+    });
+  });
+
+  it("should return invalid payload when unable to retrieve auth token", async () => {
+    mockedAxios.post.mockImplementation(async (url, data, config) => {
+      return {
+        status: 500,
+      };
+    });
+
+    const forkedGithubRepoProvider = new ForkedGithubRepoProvider();
+
+    const forkedGithubRepoProviderPayload = await forkedGithubRepoProvider.verify({
+      proofs: {
+        code,
+      },
+    } as unknown as RequestPayload);
+
+    expect(forkedGithubRepoProviderPayload).toMatchObject({ valid: false });
+  });
+
+  it("should return invalid payload when there is no id in verifyGithub response", async () => {
+    mockedAxios.get.mockImplementation(async (url, config) => {
+      return {
+        data: {
+          id: undefined,
+          login: "a-cool-user",
+          type: "User",
+        },
+        status: 200,
+      };
+    });
+
+    const forkedGithubRepoProvider = new ForkedGithubRepoProvider();
+
+    const forkedGithubRepoProviderPayload = await forkedGithubRepoProvider.verify({
+      proofs: {
+        code,
+      },
+    } as unknown as RequestPayload);
+
+    expect(forkedGithubRepoProviderPayload).toMatchObject({ valid: false });
+  });
+
+  it("should return invalid payload when the fork count for all repos is less than 1", async () => {
+    mockedAxios.get.mockImplementation(async (url, config) => {
+      return {
+        data: [
+          {
+            owner: {
+              id: "18723656",
+              type: "User",
+            },
+            fork: false,
+            forks_count: 0,
+          },
+          {
+            owner: {
+              id: "18723656",
+              type: "User",
+            },
+            fork: true,
+            forks_count: 0,
+          }
+        ],
+        status: 200,
+      };
+    });
+
+    const forkedGithubRepoProvider = new ForkedGithubRepoProvider();
+
+    const forkedGithubRepoProviderPayload = await forkedGithubRepoProvider.verify({
+      proofs: {
+        code,
+      },
+    } as unknown as RequestPayload);
+
+    expect(forkedGithubRepoProviderPayload).toMatchObject({ valid: false });
+  });
```

Comment:
This test for example (and also the others) are missing the checks if the proper AJAX requests have been made:

```suggestion
  it.only("should return invalid payload when the fork count for all repos is less than 1", async () => {
    mockedAxios.get.mockImplementation(async (url, config) => {
      return {
        data: [
          {
            owner: {
              id: "18723656",
              type: "User",
            },
            fork: false,
            forks_count: 0,
          },
          {
            owner: {
              id: "18723656",
              type: "User",
            },
            fork: true,
            forks_count: 0,
          },
        ],
        status: 200,
      };
    });

    const forkedGithubRepoProvider = new ForkedGithubRepoProvider();

    const forkedGithubRepoProviderPayload = await forkedGithubRepoProvider.verify({
      proofs: {
        code,
      },
    } as unknown as RequestPayload);

    expect(mockedAxios.get).toHaveBeenCalledTimes(2);
    expect(mockedAxios.get).toBeCalledWith(`https://api.github.com/users/${validGithubUserResponse.data.login}/repos`, {
      headers: { Authorization: `token ${validCodeResponse.data.access_token}` },
    });

    expect(forkedGithubRepoProviderPayload).toMatchObject({ valid: false });
  });
```